gammapy/catalog/core.py

```diff
@@ -51,10 +51,12 @@ class SourceCatalogObject:
     _source_name_key = "Source_Name"
     _row_index_key = "_row_index"
 
-    def __init__(self, data, data_extended=None):
+    def __init__(self, data, data_extended=None, data_spectral=None):
         self.data = Bunch(**data)
         if data_extended:
             self.data_extended = Bunch(**data_extended)
+        if data_spectral:
+            self.data_spectral = Bunch(**data_spectral)
```

Comment:
A test might be needed here...

Reply:
Yes for sure, the implementation is not finish yet !